chia-client/src/peer.rs

```diff
@@ -194,6 +309,40 @@ impl Peer {
     pub fn receiver_mut(&mut self) -> &mut broadcast::Receiver<PeerEvent> {
         &mut self.event_receiver
     }
+
+    async fn handle_inbound(
+        message: WsMessage,
+        requests: &Requests,
+        event_sender: &broadcast::Sender<PeerEvent>,
+    ) -> Result<(), ()> {
+        // Parse the message.
+        let message = Message::from_bytes(message.into_data().as_ref())?;
+
+        if let Some(id) = message.id {
+            // Send response through oneshot channel if present.
+            if let Some(request) = requests.lock().await.remove(&id) {
+                request.send(message).ok();
+            }
+            return Ok(());
+        }
+
+        macro_rules! events {
+            ( $( $event:ident ),+ $(,)? ) => {
+                match message.msg_type {
+                    $( ProtocolMessageTypes::$event => {
+                        event_sender
+                            .send(PeerEvent::$event($event::from_bytes(message.data.as_ref())?))
+                            .ok();
+                    } )+
+                    _ => {}
+                }
+            };
+        }
+
+        events!(CoinStateUpdate, NewPeakWallet);
+
```

Comment:
we don't seem to react to unknown messages being sent. I suppose we'd want to do that at some point, before production use

Reply:
Added a comment